app/views/AnalyticalFramework/utils.ts

```diff
@@ -0,0 +1,183 @@
+import { randomString } from '@togglecorp/fujs';
+import {
+    Widget,
+    ScaleWidget,
+    SingleSelectWidget,
+    MultiSelectWidget,
+    OrganigramWidget,
+    OrganigramDatum,
+    Matrix1dWidget,
+    Matrix2dWidget,
+} from '#types/newAnalyticalFramework';
+import { PartialWidget } from '#components/framework/AttributeInput';
+
+function clone(widget: PartialWidget) {
+    return ({
+        ...widget,
+        title: `${widget.title}-cloned`,
+        key: randomString(),
+        clientId: randomString(),
+        id: undefined,
```

Comment:
We should be able to re-use this logic for other widgets a well.
Something like the example below:

```javascript
// NOTE: we need to extend only the basic properties that we are modifying

function cloneWidgetSuperficially<T extends BaseWidget>(value: T): T {
    const randKey = randomString();
    return {
        ...value,
        key: randKey,
        clientId: randKey,
        id: undefined,
        title: `${value.title}: Cloned`
    };
}
```